ecr-login/cache/file.go

```diff
@@ -98,6 +98,9 @@ func (f *fileCredentialCache) fullFilePath() string {
 // file access. There is not guarantee here for handling multiple writes at once since there is no out of process locking.
 func (f *fileCredentialCache) save(registryCache *RegistryCache) error {
 	defer log.Flush()
+	if _, err := os.Stat(f.path); err != nil {
```

Comment:
I think talking about it IRL we thought that it's the file cache's responsibility to initialize itself. Given that we already use the environment variables for setting the null cache (over the file cache), it seems like this is the existing paradigm for calling, and we should stick to a common pattern. We could refactor this up later, but for the cred helper seems reasonable enough for now. We can revisit based on how the CLI experience works.